src/rpc/cli.cpp

```diff
@@ -190,6 +205,21 @@ static int cli_completion(char *token, char ***array)
    return total_matches;
 }
 
+/***
+ * @brief regex match for secret information
+ * @param source the incoming text source
+ * @returns integer 1 in event of regex match for secret information, otherwise 0
+ */
+static int cli_check_secret(const char *source)
+{
+   boost::regex expr{cli_regex_secret()};
```

Comment:
This will re-parse the regex on every invocation. This is inefficient, it should only be parsed once when ``set_regex_secret`` is called.

Reply:
ok sure ! Thanks !